File: generator/app/templates/app/src/main/java/com/example/base/baseapp/repository/remote/ExampleDataRemoteRepository.java
package com.example.base.baseapp.repository.remote;

import com.example.base.baseapp.model.ExampleData;
import com.example.base.baseapp.repository.ExampleDataRepository;

import java.util.List;

import io.reactivex.Completable;
import io.reactivex.Single;

public class ExampleDataRemoteRepository implements ExampleDataRepository {
    @Override
    public Single<List<ExampleData>> getAllExampleData() {
        //TODO: Create your network connection here
        return null;
    }

    @Override
    public Completable insertExampleData(List<ExampleData> exampleDataList) {
        return null;
    }
}

File: generators/app/templates/app/src/main/kotlin/com/example/base/baseapp/repository/remote/ExampleDataRemoteRepository.kt
package com.example.base.baseapp.repository.remote

import com.example.base.baseapp.model.ExampleData
import com.example.base.baseapp.repository.ExampleDataRepository

import io.reactivex.Completable
import io.reactivex.Single

//TODO: Override allExampleData and insertExampleData and then remove abstract modifier
abstract class ExampleDataRemoteRepository : ExampleDataRepository {
    //override val allExampleData: Single<List<ExampleData>>
    //get() = TODO: Create your network connection here

    //override fun insertExampleData(exampleDataList: List<ExampleData>): Completable {
    //TODO: Create your network connection here
    //}
}
